### PR TITLE
:sparkles: addon add support for multiple same-GVK

### PR DESCRIPTION
**Pull request by JoaoBraveCoding**
## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by qiujian16** [comment]
cc @haoqing0110 who is working on this.

File: pkg/addon/controllers/addonconfiguration/graph.go [hunk 1]
```diff
@@ -69,24 +69,26 @@ func (n *addonNode) setRolloutStatus() {
...
-					n.status.Status = clustersdkv1alpha1.Failed
-					n.status.LastTransitionTime = &progressingCond.LastTransitionTime
-				case addonv1alpha1.ProgressingReasonProgressing:
-					n.status.Status = clustersdkv1alpha1.Progressing
-					n.status.LastTransitionTime = &progressingCond.LastTransitionTime
-				default:
-					n.status.Status = clustersdkv1alpha1.Progressing
+		if desiredArray, ok := n.desiredConfigs[actual.ConfigGroupResource]; ok {
+			for _, desired := range desiredArray {
+				// desired config spec hash doesn't match actual, set to ToApply
```
Comment (haoqing0110) [inline pkg/addon/controllers/addonconfiguration/graph.go:74]: Need to compare the `desired.ConfigGroupResource` and `actual.ConfigGroupResource`.

**Comment by haoqing0110** [comment]
The overall looks good, some comparisons need to consider the ConfigReferent as well. Based on our previous discussion, this should have minimal impact/change to support multiple configs, the graph key is still GVK, and the value changes from a single config to an array.

**Comment by haoqing0110** [comment]
PR https://github.com/open-cluster-management-io/ocm/pull/585 is based on @JoaoBraveCoding 's effort and adds more testing case.

**Pull request closed without merging**